### PR TITLE
HPCC-12976 Add xml only output option

### PR DESCRIPTION
**Pull request by garonsky**
Signed-off-by: Gleb Aronsky gleb.aronsky@lexisnexis.com

### CODE REVIEW

**Comment by garonsky** [comment]
@stuartort please review

**Comment by garonsky** [comment]
@stuartort renamed the variable.  Please review.

**Comment by stuartort** [comment]
looks good

**Comment by garonsky** [comment]
Changes have been squashed.

@richardkchapman this is ready to merge.

File: tools/testsocket/testsocket.cpp [hunk 1]
```diff
@@ -538,13 +539,23 @@ int doSendQuery(const char * ip, unsigned port, const char * base)
 
         if (trace != NULL)
         {
-            fprintf(trace, "query: %s\n", query);
-            if (saveResults)
-                fprintf(trace, "result: %s\n", result.str());
+            if (rawOnly == false)
```
Comment (richardkchapman) [inline tools/testsocket/testsocket.cpp]: if (!rawOnly) would be more normal...

File: tools/testsocket/testsocket.cpp [hunk 2]
```diff
@@ -538,13 +539,23 @@ int doSendQuery(const char * ip, unsigned port, const char * base)
...
+                if (saveResults)
+                    fprintf(trace, "result: %s\n", result.str());
+            }
+            else
+            {
+                fprintf(trace, "%s", result.str());
+            }
 
-            if (showTiming)
+            if (showTiming && rawOnly == false)
```
Comment (richardkchapman) [inline tools/testsocket/testsocket.cpp]: As above

File: tools/testsocket/testsocket.cpp [hunk 3]
```diff
@@ -936,10 +954,13 @@ int main(int argc, char **argv)
     endtime = get_cycles_now();
     if (!justResults)
     {
-        if (trace != NULL)
+        if (rawOnly == false)
```
Comment (richardkchapman) [inline tools/testsocket/testsocket.cpp]: As above

**Comment by richardkchapman** [comment]
Minor comments only - I will still merge

**Comment by richardkchapman** [comment]
Can you check the issue number - the one referenced does not look related.

**Comment by richardkchapman** [comment]
Ok, I see how it is related. Please use a separate JIRA for this sort of thing though.

**Pull request merged**